### ISSUE TITLE
Error due to file-type import when compiling with SWC

### ISSUE THREAD

**Issue by xiniria** [ForestAdmin/agent-nodejs#757]
Hi there,
I'm using Forest Admin on a NestJS app, with the dedicated integration that allows to run a Forest Admin backend in very few lines of code. Recently, I saw that NestJS introduced (in version 10) [support for SWC as the compiler](https://docs.nestjs.com/recipes/swc) (instead of the default `tsc`). I tried it, but stumbled upon a problem: when compiling, I get this error:

```
/Users/.../node_modules/@forestadmin/datasource-customizer/dist/decorators/binary/collection.js:7
const file_type_1 = __importDefault(require("file-type"));
                                    ^
Error [ERR_REQUIRE_ESM]: require() of ES Module /Users/.../node_modules/file-type/index.js from /Users/.../node_modules/@forestadmin/datasource-customizer/dist/decorators/binary/collection.js not supported.
Instead change the require of index.js in /Users/.../node_modules/@forestadmin/datasource-customizer/dist/decorators/binary/collection.js to a dynamic import() which is available in all CommonJS modules.
```

After investigation, I noticed that installing `@swc/cli` and `@swc/core` triggered the update of package `file-type` to version `17.1.6`, while `@forestadmin/datasource-customizer` specifies version `version-16` as a dependency, so I'm guessing this could be part of the problem.

If I update the file `@forestadmin/datasource-customizer/dist/decorators/binary/collection.js` manually to replace `require("file-type")` with `import("file-type")`, the compilation works as expected.

I'm no expert but I think it might possibly be linked to this `file-type` version; perhaps updating it in the `datasource-customizer` package would solve the problem. If it would, I will be happy to submit a PR to solve this.

## Expected behavior

The compilation succeeds with SWC.

## Actual behavior

The compilation fails with SWC.

## Failure Logs

```
/Users/.../node_modules/@forestadmin/datasource-customizer/dist/decorators/binary/collection.js:7
const file_type_1 = __importDefault(require("file-type"));
                                    ^
Error [ERR_REQUIRE_ESM]: require() of ES Module /Users/.../node_modules/file-type/index.js from /Users/.../node_modules/@forestadmin/datasource-customizer/dist/decorators/binary/collection.js not supported.
Instead change the require of index.js in /Users/.../node_modules/@forestadmin/datasource-customizer/dist/decorators/binary/collection.js to a dynamic import() which is available in all CommonJS modules.
```

## Context

- Package Version: 1.11.0
- `@nestjs/core` version: 10.0.0
- `@swc/core` version: 1.3.66

**Comment by charlelie-borella** [comment]
I have the same issue and the same config as you @xiniria.

**Comment by jeffladiray** [comment]
Hey @charlelie-borella @xiniria :wave:

I think you should be able to test this via overriding the dependency version.
```
  "overrides": {
    "@forestadmin/datasource-customizer": {
      "file-type": "17.1.6"
    }
  }
```

In the meantime, I'll give it a shot on my end to check as well

**Comment by jeffladiray** [comment]
Eventually, we released #758 that could potentially fix the mentionned issue.

Could you try to upgrade to `@forestadmin/agent: 1.13.3`? 🙏 

Let us know !

**Comment by xiniria** [comment]
It works, thanks a lot!
